### DISCUSSION TITLE
Keine Verbindung zum Wechselrichter -> Serial Monitor Bericht

### DISCUSSION THREAD

**Discussion by Flip9Flip** [tbnobody/OpenDTU#1133] [Q&A] [answered]
Grüß Euch,

soweit hat alles geklappt(dachte ich zumindest). Leider bekomme ich keine Verbindung zum Wechselrichter. Der Serial Monitor zeigt folgendes:

14:13:30.890 > 2019 12:21:.���� (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
14:13:30.890 > configsip: 0, SPIWP:0xee
14:13:30.890 > clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
14:13:30.890 > mode:DIO, clock div:2
14:13:30.890 > load:0x3fff0030,len:1184
14:13:30.890 > load:0x40078000,len:13192
14:13:30.890 > load:0x40080400,len:3028
14:13:30.890 > entry 0x400805e4
14:13:30.890 > E (626) esp_core_dump_flash: No core du枀�partition found!
14:13:30.890 > E (627) esp_core_dump_flash: No core dump partition found!
14:13:30.890 > 
14:13:30.890 > Starting OpenDTU
14:13:30.890 > Initialize FS... done
14:13:30.890 > Reading configuration... done
14:13:30.890 > Reading PinMapping... [   174][E][vfs_api.cpp:105] open(): /littlefs/pin_mapping.json does not exist, no permits for creation
14:13:30.890 > using default config done
14:13:30.890 > Initialize Network... done
14:13:30.890 > Setting Hostname... Configuring WiFi STA using new credentials... done
14:13:30.890 > Initialize NTP... done
14:13:30.890 > Initialize SunPosition... done
14:13:30.890 > Initialize MqTT... done
14:13:30.890 > Initialize WebApi... done
14:13:30.890 > Initialize Display... done
14:13:30.890 > Initialize LEDs... done
14:13:30.890 > Check for default DTU serial... done
14:13:30.890 > Initialize Hoymiles interface... Invalid pin config
14:13:30.890 > Switch to WiFi mode
14:13:30.890 > Setting Hostname... done
14:13:30.891 > Configuring WiFi STA using new credentials... WiFi disconnected
14:13:30.891 > Try reconnecting
14:13:30.891 > Network lost connection
14:13:30.891 > E (1168) wifi:sta is connecting, return error
14:13:30.891 > [   385][E][WiFiSTA.cpp:278] begin(): connect failed!
14:13:30.891 > done
14:13:30.891 > Configuring WiFi STA DHCP IP... done
14:13:30.891 > WiFi connected
14:13:30.891 > WiFi got ip: 192.168.178.48
14:13:30.891 > Network connected
14:14:05.540 > Websocket: [/livedata][1] connect
14:14:05.553 > Websocket: [/livedata][2] connect
14:16:30.157 > Admin mode disabled

So wie das aussieht, verbergen sich hier mehrer Fehlermeldungen (core not found, lost network connection). Wäre/könnte einer so nett sein und mir ein bisschen Licht in's dunkle bringen, an was das liegen könnte? Wäre wirklich schön die DTU zum laufen zu bringen:) 

Vielen Dank vorab!

**Comment by tbnobody** [comment]
Wichtig und entscheidend ist nur das  hier: 
```
14:13:30.890 > Reading PinMapping... [ 174][E][vfs_api.cpp:105] open(): /littlefs/pin_mapping.json does not exist, no permits for creation
14:13:30.890 > Initialize Hoymiles interface... Invalid pin config
```

Das bedeutet, dass du für dein Board noch die passende [JSON Datei](https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles) hochladen musst, damit die Firmware weiß an welchen Pins das NRF Modul angeschlossen ist.

**Reply by Flip9Flip** [reply]
Danke für die schnelle Antwort! 
Bedeutet, ich wähle die PinMapping.cpp Datei, dann öffnet sich VSC und dort füge ich die komplette Zeile (aus deinem Link) einfach ein? und lade alles erneut auf die ESP?

**Reply by Flip9Flip** [reply]
oder soll ich hier: 
PinMappingClass::PinMappingClass()
{
    memset(&_pinMapping, 0x0, sizeof(_pinMapping));
    _pinMapping.nrf24_clk = HOYMILES_PIN_SCLK;
    _pinMapping.nrf24_cs = HOYMILES_PIN_CS;
    _pinMapping.nrf24_en = HOYMILES_PIN_CE;
    _pinMapping.nrf24_irq = HOYMILES_PIN_IRQ;
    _pinMapping.nrf24_miso = HOYMILES_PIN_MISO;
    _pinMapping.nrf24_mosi = HOYMILES_PIN_MOSI;

einfach die z.b.: "18" Pin Belegung hinterlegen?

**Reply by tbnobody** [reply]
Nein, du sollst garakeinen Sourcecode ändern, sondern so wie in meinem [Link](https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles) beschrieben eine fertige JSON Datei über das Webinterface der DTU hochladen.

**Reply by Flip9Flip** [reply]
16:08:32.099 > Reading configuration... done
16:08:32.099 > Reading PinMapping... using default config don
jedoch immer noch
16:08:32.099 > Initialize Hoymiles interface... Invalid pin config

vermutlich weil ich nicht finde, wo genau ich auf dem Webinterface das Geräteprofil ändern kann.. 

Tut mir leid für die Umstände und vielen Dank für die Unterstützung!

**Reply by tbnobody** [reply, marked as answer]
> To change the device profile, navigate to the "Device Manager" and selected the appropriate profile. 

Einstellungen --> Hardware und dann hier das gewünschte Profil wählen
![image](https://github.com/tbnobody/OpenDTU/assets/1753222/9e062b0c-c0ab-40b6-b53b-f1b33e9b45ab)

**Reply by Flip9Flip** [reply]
Wollt mich noch herzlich bedanken. Hat alles wunderbar geklappt!
